biogeochem/EDCanopyStructureMod.F90

```diff
@@ -469,98 +485,150 @@ subroutine DemoteFromLayer(currentSite,currentPatch,i_lyr)
          ! If this is probabalistic demotion, we need to do a round of normalization.
          ! And then a few rounds where we pre-calculate the demotion areas
          ! and adjust things if the demoted area wants to be greater than
-         ! what is available.  sumweights_old is used to sum up the exclusion
-         ! area of those cohorts that have not reached their total canopy area 
-         ! yet.
-         
+         ! what is available. The math is too hard to explain here, see
+         ! the tech note section on promotion/demotion.
+
          if (ED_val_comp_excln .ge. 0.0_r8 ) then
             
-            remainder_area = 0.0_r8
-            sumweights_old = 0.0_r8
+            scale_factor_min  = 1.e10_r8
+            scale_factor      = 0._r8
             currentCohort => currentPatch%tallest
             do while (associated(currentCohort))
 
                if(currentCohort%canopy_layer  ==  i_lyr) then 
-                  currentCohort%excl_weight = demote_area*currentCohort%excl_weight/sumweights
-                  if( currentCohort%excl_weight > currentCohort%c_area ) then
-                     remainder_area = remainder_area + currentCohort%excl_weight-currentCohort%c_area
-                     currentCohort%excl_weight = currentCohort%c_area
-                  else
-                     sumweights_old = sumweights_old + currentCohort%excl_weight
-                  end if
+
+                  currentCohort%excl_weight = currentCohort%excl_weight/sumweights
```

Comment:
Do I understand correctly here that the code will still run through the normal (stochastic) case first?  Otherwise ```sumweights``` would start off as ```0.0_r8``` and result in a divide-by-zero?

Reply:
I think I answered my own question: this has to be true given that the ```currentCohort%excl_weight``` is also computed in that section as well.

Reply:
I see; we need to run through the existing weights first to get their sum and then run back through them again to normalize.